### PR TITLE
[WIP] Added Withdraw Coin Daemon

### PR DESCRIPTION
**Pull request by dinesh-skyach**
Added Withdraw Coin Daemon

### CODE REVIEW

File: lib/wallet_client/geth.rb [hunk 1]
```diff
@@ -17,6 +17,59 @@ def create_address!(options = {})
       end
     end
 
+    def load_balance!(currency)
+      PaymentAddress
```
Comment (ysv) [inline lib/wallet_client/geth.rb:21]: No peatio specific code in wallet_client

Comment (ysv) [reply lib/wallet_client/geth.rb:21]: Move this to service

File: lib/wallet_client/geth.rb [hunk 2]
```diff
@@ -43,5 +96,48 @@ def json_rpc(method, params = [])
...
+    def valid_address?(address)
+      address.to_s.match?(/\A0x[A-F0-9]{40}\z/i)
+    end
+
+    def valid_txid?(txid)
+      txid.to_s.match?(/\A0x[A-F0-9]{64}\z/i)
+    end
+
+    def contract_address(currency)
+      normalize_address(currency.erc20_contract_address)
```
Comment (ysv) [inline lib/wallet_client/geth.rb:139]: we don't have currency in WalletClient

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/workers/worker/withdraw_coin.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          Rails.logger.warn { "The destination address doesn't exist. Skipping..." }
+          withdraw.fail!
+          return
+        end
+
+        Rails.logger.warn { "Information: sending #{withdraw.amount.to_s("F")} (exchange fee is #{withdraw.fee.to_s("F")}) #{withdraw.currency.code.upcase} to #{withdraw.rid}." }
+
+        currency = withdraw.currency
+
+        wallet = Wallet.active.deposit.where(blockchain_key: currency.blockchain_key).first
```
Comment (ysv) [inline app/workers/worker/withdraw_coin.rb:34]: Only withdraw wallets here

**Review by ysv — CHANGES_REQUESTED** [review]

**Comment by dinesh-skyach** [comment]
moved #1485

**Pull request closed without merging**